### DISCUSSION TITLE
Dashboard questions

### DISCUSSION THREAD

**Discussion by verisgit** [basking-in-the-sun2000/solar-logger#38] [General]
Just opening this one to track some questions about the dashboard that I assume some others may have when setting this up.

![image](https://user-images.githubusercontent.com/18733413/165441332-811b4f6e-c471-477d-8c61-8bc2c6eee8b4.png)

It's been collecting for a few days now. Mostly working well and seems stable. The timezone seems to be incorrectly offset but I'll look into that along with some of your comments from the other ticket.

Do you have any brief descriptions of what some of the fields/charts are displaying? In particular the Flowchart values/arrows, dials and forecast chart.

**Comment by verisgit** [comment]
I couldn't see any further timezone changes in Grafana either in the JSON or settings, they all seem to be correct.

**Comment by verisgit** [comment]
I also noticed grid is missing from this chart, when I looked at the data it's null. Does yours work? Maybe I have an outdated query.
![image](https://user-images.githubusercontent.com/18733413/165567250-8ce87c93-b908-4bbd-96bb-89770bc8859e.png)

**Comment by basking-in-the-sun2000** [comment]
Moved it to a discussion, since I figured others might have similar doubts. 

This general panel is designed for 1 day of displayed data. Usually use Today or Yesterday. It gets a bit confused if you set it for longer periods.

1) Been working on moving the timezone adjustment to one place, since grafana is a bit difficult as to how it handles it. First, you need to select the one for the window. You can use your location or browser. 
<img width="533" alt="Screen Shot 2022-04-27 at 15 02 12" src="https://user-images.githubusercontent.com/61602111/165628810-399add38-f16c-4b25-9169-d96551f93bb6.png">

<img width="146" alt="Screen Shot 2022-04-27 at 15 02 46" src="https://user-images.githubusercontent.com/61602111/165629104-b1f884be-b22e-48bd-9045-4c9d7ebb21af.png">


The other one is in the settings for the main dashboard (gear icon on the top menu). 
<img width="291" alt="Screen Shot 2022-04-27 at 15 00 48" src="https://user-images.githubusercontent.com/61602111/165628901-5b1eb3f9-ff57-4db3-805f-53268e3a39c4.png">

I been using UTC for many defaults in the panels. If you search the json for UTC, you can see where these are. Been trying to move them over to use the value of the variable.

Open an issue, if after setting these your graphs are still wrong. I can force grafana to use the variable for timezone

2) The colors are according to your site. You determine how much is acceptable to you. My situation (and main driving design) might not apply to you, since I'm on a yearly balance plan (net-metering?). I have a year to use my excess energy. 

To select the colors you want, you set them in the threshold for that item. This is an example of the peak_max in Today's panel. Each item with color has its settings. In my case, an export of 5 kWh a day is low, and thus the red coloring. Seems that grafana doesn't allow the use of variables for thresholds, so no easy way to fix this :(
<img width="373" alt="Screen Shot 2022-04-27 at 15 07 27" src="https://user-images.githubusercontent.com/61602111/165629021-6c730872-8395-4a93-a85f-8738edd707d0.png">


3) Production balance shows your energy balance between bills. I'm on a bi-monthly billing scheme, and the reading is on the 18th. Thus, those values are hardcoded, you can change them in the panel. Grafana doesn't let you limit the displayed data to a date taken from another query, so at the beginning of the period, it shows part of the balance of the last period (which gets better the closer to your next bill). Either you don't have enough data for this graph to display, or you haven't set up the parameters in the config.py?

4) Most of the items in today's panel (flowchart) have a description (albeit maybe a bit sparse) if you hover over the item for a moment. There are also links to other pages (the little icon on the top left of the panels) and some have a description (the little _i_ on the top left of the panel)

5) The weird little icons on the top right and the weird graph in your last post, are related. See the config file. These are for load diverting. You could have the rpi output a signal to turn on or off a load (through a relay or contactor). The graph was an internal graph to try to reduce the jitteriness of the logic. Works nicely on cloudless days, but on a cloudy day…


This is the way it looks for me

<img width="1335" alt="Screen Shot 2022-04-27 at 14 12 23" src="https://user-images.githubusercontent.com/61602111/165629201-f9d03188-92ef-4a4b-913d-189285b68a33.png">

**Comment by basking-in-the-sun2000** [comment]
Here are a couple of other screenshots to show different times of the day. As you can see several of the indicators change color depending on their values. So you just need to change the thresholds to values that make sense to your site.

At night
<img width="1335" alt="Screen Shot 2022-04-27 at 23 01 03" src="https://user-images.githubusercontent.com/61602111/165676544-a5f26987-eede-4d2b-954f-470bfc252c30.png">

Yesterday
<img width="1337" alt="Screen Shot 2022-04-27 at 17 09 40" src="https://user-images.githubusercontent.com/61602111/165676614-4def559d-d800-419d-937a-9414e994fba6.png">
